### PR TITLE
Allow NotificationEmail to receive a config

### PR DESCRIPTION
**Pull request by Thrillberg**
## Summary
This PR allows `NotificationEmail` to receive a config as an argument. This can sometimes include a newly necessary attribute, `lighthouse_updated_at`.

## Related issue(s)
https://app.zenhub.com/workspaces/vagov-product-team-forms-634853151f5c6000165942bc/issues/gh/department-of-veterans-affairs/va.gov-team-forms/1668

### CODE REVIEW

File: modules/simple_forms_api/app/services/simple_forms_api/notification_email.rb [hunk 1]
```diff
@@ -21,10 +21,11 @@ class NotificationEmail
...
 
-    def initialize(form_data:, form_number:, confirmation_number:, notification_type: :confirmation, user: nil)
-      @form_data = form_data
-      @form_number = form_number
-      @confirmation_number = confirmation_number
+    def initialize(config, notification_type: :confirmation, user: nil)
+      @form_data = config[:form_data]
+      @form_number = config[:form_number]
+      @confirmation_number = config[:confirmation_number]
+      @lighthouse_updated_at = config[:lighthouse_updated_at]
```
Comment (pennja) [inline modules/simple_forms_api/app/services/simple_forms_api/notification_email.rb:28]: It looks like this variable is never used. Do we need it?

File: modules/simple_forms_api/app/services/simple_forms_api/notification_email.rb [hunk 2]
```diff
@@ -21,10 +21,11 @@ class NotificationEmail
...
       @notification_type = notification_type
```
Comment (pennja) [inline modules/simple_forms_api/app/services/simple_forms_api/notification_email.rb:29]: Given that this is only used in one spot, does it make more sense to set the instance variable for the one that uses it?

```ruby
@template_id = TEMPLATE_IDS[form_number][notification_type]
```

File: modules/simple_forms_api/app/services/simple_forms_api/notification_email.rb [hunk 3]
```diff
@@ -21,10 +21,11 @@ class NotificationEmail
...
     SUPPORTED_FORMS = TEMPLATE_IDS.keys
 
-    def initialize(form_data:, form_number:, confirmation_number:, notification_type: :confirmation, user: nil)
-      @form_data = form_data
-      @form_number = form_number
-      @confirmation_number = confirmation_number
+    def initialize(config, notification_type: :confirmation, user: nil)
+      @form_data = config[:form_data]
+      @form_number = config[:form_number]
+      @confirmation_number = config[:confirmation_number]
```
Comment (pennja) [inline modules/simple_forms_api/app/services/simple_forms_api/notification_email.rb:27]: The consolidation is nice and necessary given our linter settings, but we need some sort of validation to ensure the initialization will fail gracefully and clearly when one or multiple of these parameters are not provided.

File: modules/simple_forms_api/spec/services/notification_email_spec.rb [hunk 4]
```diff
@@ -337,17 +316,16 @@
...
     it 'sends the confirmation email' do
       allow(VANotify::EmailJob).to receive(:perform_async)
 
-      subject = described_class.new(
-        form_data: data,
-        form_number: 'vba_21_0966',
-        confirmation_number: 'confirmation_number',
-        user:
-      )
+      subject = described_class.new(config, user:)
```
Comment (pennja) [inline modules/simple_forms_api/spec/services/notification_email_spec.rb:328]: I like the consolidation in all these tests!

File: modules/simple_forms_api/spec/services/notification_email_spec.rb [hunk 5]
```diff
@@ -12,6 +12,10 @@
         )
         JSON.parse(fixture_path.read)
       end
+      let(:config) do
+        { form_data: data, form_number: 'vba_21_10210',
+          confirmation_number: 'confirmation_number' }
+      end
```
Comment (pennja) [inline modules/simple_forms_api/spec/services/notification_email_spec.rb:19]: It would be prudent to also have test coverage for when we provide invalid or nil config parameters as well.

File: modules/simple_forms_api/app/services/simple_forms_api/notification_email.rb (hunk 2 above)
Comment (Thrillberg) [reply modules/simple_forms_api/app/services/simple_forms_api/notification_email.rb:29]: The `send` method does a check for `form_number` before this line so I'd rather not make this change since that would complicate the initializer, I think.

Comment (pennja) [reply modules/simple_forms_api/app/services/simple_forms_api/notification_email.rb:30]: Fair enough!

File: modules/simple_forms_api/app/services/simple_forms_api/notification_email.rb [hunk 6]
```diff
@@ -47,6 +49,11 @@ def send
 
     private
 
+    def check_missing_keys(config)
+      missing_keys = %i[form_data form_number confirmation_number].select { |key| config[key].nil? }
```
Comment (pennja) [inline modules/simple_forms_api/app/services/simple_forms_api/notification_email.rb:53]: So sorry, but I think we'd probably want this instead:

```suggestion
      missing_keys = %i[form_data form_number confirmation_number].select { |key| config[key] && config[key].nil? }
```

**Review by pennja — APPROVED** [review]
Looks good, great work! 👍
